### PR TITLE
fix: Allow more than 10 items in sourceData of menu button inside table widget

### LINKED ISSUE #24504: [Bug]: Dynamic menu buttons in tables break if there are more than 10 rows
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I have a table, and one of the columns is a menu button. This menu button is dynamic. In this menu button, I have hard-coded a single option. The options will be the same for each row. This works fine for up to ten table rows. When I add the eleventh table row, the menu buttons no longer open. We get this error in the table:

image

I expect it only to allow 10 menu options per row instead of 10 sets of menu options in the table. Here it appears to be checking the length of the wrong arrray. Each button only has one option, but there are more than 10 buttons.

### Steps To Reproduce

1. Open the public sample app of this issue.
2. Click a menu button in the table.
3. Observe that the menu appears.
4. Click the `Add One Row` button.
5. Click a menu button in the table again.
6. Observe that nothing happens.

### Public Sample App

https://app.appsmith.com/app/table-menu-button-issue/example-648b30f7e923f357ae1e8ec1

### Environment

Production

### Issue video log

_No response_

### Version

Cloud v1.9.22-SNAPSHOT

Front logo [Front conversations](https://app.frontapp.com/open/top_fafrg)

### PR DESCRIPTION
**Pull request by dhruvikn**
## Description
The `sourceData` for the menu button allows a max of 10 menu items. This PR uses the inbuilt appsmith validations to check if the source data of menu buttons when in the table widget is - 
- an array
    - having a child array with allowed types = TEXT, NUMBER, and OBJECT

#### PR fixes following issue(s)
Fixes #24504

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/widgets/MenuButtonWidget/validations.ts [hunk 1]
```diff
@@ -32,7 +32,20 @@ export function sourceDataArrayValidation(
       let isValid = true;
       let message = { name: "", message: "" };
 
-      if (options.length > 10) {
+      if (props.type === "TABLE_WIDGET_V2") {
```
Comment (sbalaji1192) [inline app/client/src/widgets/MenuButtonWidget/validations.ts:35]: This doesn't seem like a good place to keep the validation related to the table widget v2.

**Review by sbalaji1192 — APPROVED** [review]

**Comment by dhruvikn** [comment]
A note to whoever picks this up for QA - we'll need to test both the menu button widget's sourceData validations and when the menu button widget is in the table widget.

cc: @laveena-en @chandannkumar

**Comment by chandannkumar** [comment]
Tested this PR and LGTM

**Pull request merged**